addons/source-python/packages/source-python/players/engines/orangebox/tf.py

```diff
@@ -0,0 +1,49 @@
+# ../players/engines/orangebox/cstrike.py
+
+"""Provides Team Fortress 2 specific Player based functionality."""
+
+# =============================================================================
+# >> IMPORTS
+# =============================================================================
+# Source.Python Imports
+from . import Player as _Player
+from configobj import ConfigObj
+from core import GAME_NAME
+from paths import SP_DATA_PATH
+
+# =============================================================================
+# >> CLASSES
+# =============================================================================
+class Player(_Player):
+    caching = True
+    
+    # Team
+    _team_data = ConfigObj(SP_DATA_PATH / 'teams' / GAME_NAME + '.ini', unrepr=True)
+
+    _teams_names = _team_data.get('names')
+    _class_names = _team_data.get('classes')
+
+    @property
+    def team_name(self):
+        teams_by_number = dict(zip(self._teams_names.values(), self._teams_names.keys()))
+        return teams_by_number[self.team]
+    
+    @team_name.setter
+    def team_name(self, team_name):
+        if team_name not in self._teams_names:
+            raise ValueError(f"Invalid team_name: {team_name}")
+        
+        self.team = self._teams_names[team_name]
+    
+    @property
+    def player_class_name(self):
+        class_by_number = dict(zip(self._class_names.values(), self._class_names.keys()))
```

Comment:
This is redundant to rebuild that mapping every calls, and it should really just be globalized into the `players.teams` module like the teams mapping are.